tests/unit/test_movielens.py

```diff
@@ -1,8 +1,11 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
so right now we are mocking the download behavior and are not downloading 100k and 1M anymore. How can we check that these datasets still work?

Reply:
One idea could be to check that the files exist (without download them) and then use the smoke and integration tests to actually download them and test them. Maybe we can move 100k and 1M to smoke and 10 and 20M to integration. 

Any other ideas?

Reply:
do any smoke tests use the 1M dataset? ideally not, in which case I would be fine moving the 100k test which downloads the data to smoke and the larger ones to integration.

the main thing is I want to be able to run the unit test suite around 1 min or less, and I'd like to be able to do it if I don't have internet access.

with this mocking we will ensure that load_pandas_df and load_spark_df is doing the right thing if we give it properly formatted files (at least for 100k, and 1M formats), if that format changes on MovieLens side then this particular test wouldn't catch it (however integration tests should)

Reply:
I checked and as you said we are only using 100k on smoke, so this makes sense:

> moving the 100k test which downloads the data to smoke and the larger ones to integration.

Reply:
k, reverting changes

Reply:
About lack of internet access, would it make sense if there was a marker so that you could have the option to run the mock tests with this flag and otherwise run the tests we had before?
Also, why are there 3 samples for the Spark test and 100000 for the Pandas test?